README.md

```diff
@@ -1,9 +1,94 @@
 
-A sidecar container that
+This is necessary because Kubelet is responsible for issuing CSI `NodeGetInfo`,
+`NodeStageVolume`, `NodePublishVolume` calls. The `node-driver-registrar` registers
+your CSI driver with Kubelet so that it knows which Unix domain socket to issue
+the CSI calls on.
+
+## Compatibility
+Latest stable release | Branch| Compatible with CSI Version | Container Image | Min K8s Version | Max K8s Version
+-----------------------|-----------------------|-----------------------------|----------------|---------|--------|
+[node-driver-registrar v1.0.1](https://github.com/kubernetes-csi/cluster-driver-registrar/releases/tag/v1.0.1) | [release-1.0](https://github.com/kubernetes-csi/node-driver-registrar/tree/release-1.0) | [CSI Spec v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | quay.io/k8scsi/csi-cluster-driver-registrar:v1.0.1 | 1.13 | -
+[driver-registrar v0.4.2](https://github.com/kubernetes-csi/driver-registrar/releases/tag/v0.4.2) | [release-0.4](https://github.com/kubernetes-csi/driver-registrar/tree/release-0.4) | [CSI Spec v0.3.0](https://github.com/container-storage-interface/spec/releases/tag/v0.3.0) | quay.io/k8scsi/driver-registrar:v0.4.2 | 1.10 | -
+
+
+For release-0.4 and below, please refer to the [node-registrar
+repository](https://github.com/kubernetes-csi/driver-registrar)
+
+## Usage
+There are two UNIX domain sockets used by the node-driver-registrar:
+
+* Registration socket:
+  * Registers the driver with kubelet.
+  * Created by the `node-driver-registrar`.
+  * Exposed on a Kubernetes node via hostpath in the Kubelet plugin registry
+    (typically `/var/lib/kubelet/plugins_registry/<drivername.example.com>-reg.sock`.
+
+* CSI driver socket:
+  * Used by kubelet to interact with the CSI driver
+  * Created by the CSI driver
+  * Exposed on a Kubernetes node via hostpath somewhere other than the Kubelet
+    plugin registry (typically `/var/lib/kubelet/plugins/<drivername.example.com>/csi.sock`.
+  * This is the socket referenced by the `--csi-address` and `--kubelet-registration-path` arguments.
+
+### Required arguments
+
+* `--csi-address`: This is the path to the CSI driver socket (defined above) inside the
+  pod that the `node-driver-registrar` container will use to issue CSI
+  operations (e.g. `/csi/csi.sock`).
+* `--kubelet-registration-path`: This is the path to the CSI driver socket on
+  the host node that kubelet will use to issue CSI operations (e.g.
+  `/var/lib/kubelet/plugins/<drivername.example.com>/csi.sock).
+* `KUBE_NODE_NAME` environment variable: This is the name of the Kubernetes Node.
+
+### Required permissions
+The node-driver-registrar does not interact with the Kubernetes API, so no RBAC
+rules are needed.
+
+It does, however, need to be able to mount hostPath volumes and have the file
+permissions to:
+* read from `/var/lib/kubelet/plugins/<drivername.example.com>/`
+* write to `/var/lib/kubelet/plugins_registry/`
```

Comment:
Based on our discussion the only thing that might be needed by `node-driver-registrar` is:

> * Read and write to the registration socket (typically in the `/var/lib/kubelet/plugins_registry/` directory)
>   * Used by the `node-driver-registrar` to register the driver with Kubelet.
> * Read access to the driver socket (typically in the `/var/lib/kubelet/plugins/<drivername.example.com>/` directory)
>   * Used by the `node-driver-registrar` to fetch the driver name from the driver container (via a `GetPluginInfo(...)` CSI call).